src/instructlab/rag/haystack/components/document_splitter.py

```diff
@@ -0,0 +1,101 @@
+# Standard
+from typing import Any, Dict, List, cast
+import logging
+
+# Third Party
+from docling_core.transforms.chunker.hybrid_chunker import HybridChunker
+from docling_core.types import DoclingDocument
+from docling_core.types.legacy_doc.document import (
+    ExportedCCSDocument as LegacyDoclingDocument,
+)
+from docling_core.utils.legacy import legacy_to_docling_document
+from haystack import Document, component  # type: ignore
+from haystack.core.serialization import (  # type: ignore
+    default_from_dict,
+    default_to_dict,
+)
+from pydantic_core._pydantic_core import ValidationError
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class DoclingDocumentSplitter:
+    SUPPORTED_CONTENT_FORMATS = ["json"]
+
+    def __init__(self, embedding_model_id=None, content_format=None, max_tokens=None):
+        self.__chunker = HybridChunker(
+            tokenizer=embedding_model_id, max_tokens=max_tokens
+        )
+        self.__embedding_model_id = embedding_model_id
+
+        if content_format not in self.SUPPORTED_CONTENT_FORMATS:
+            raise ValueError(
+                f"Only the following input formats are currently supported: {self.SUPPORTED_CONTENT_FORMATS}."
+            )
+        self.__content_format = content_format
+
+    @component.output_types(documents=List[Document])
+    def run(self, documents: List[Document]):
+        if not isinstance(documents, list) or (
+            documents and not isinstance(documents[0], Document)
+        ):
+            raise TypeError(
+                "DoclingDocumentSplitter expects a List of Documents as input."
+            )
+
+        split_docs = []
+        for doc in documents:
+            if doc.content is None:
+                raise ValueError(f"Missing content for document ID {doc.id}.")
+
+            chunks = self._split_with_docling(doc.meta["file_path"], doc.content)
+            current_split_docs = [Document(content=chunk) for chunk in chunks]
+            split_docs.extend(current_split_docs)
+
+        return {"documents": split_docs}
```

Comment:
this feels like it should be a Class not just a dict

Reply:
This is how haystack works, can’t do differently